### PR TITLE
Kumarde/fill out fields

### PR DESCRIPTION
**Pull request by kumarde**
Filling out outstanding field required for cert-daemon

### CODE REVIEW

File: lints/base.go [hunk 1]
```diff
@@ -207,6 +207,11 @@ type LintReport struct {
 	ERsaNoPublicKey                                      ResultStruct `json:"e_rsa_no_public_key,omitempty"`
 	ESubCertCertificatePoliciesMissing                   ResultStruct `json:"e_sub_cert_certificate_policies_missing,omitempty"`
 	ESubCertKeyUsageCrlSignBitSet                        ResultStruct `json:"e_sub_cert_key_usage_crl_sign_bit_set,omitempty"`
+	InfosPresent                                         bool         `json:"infos_present,omitempty"`
```
Comment (zakird) [inline lints/base.go]: These should be in `ZLintResult`...

File: lints/base.go [hunk 2]
```diff
@@ -216,10 +221,24 @@ func (report *LintReport) Execute(cert *x509.Certificate) error {
 			return err
 		}
 		l.updateReport(report, res)
+		updateErrorStatePresent(report, res)
 	}
 	return nil
 }
 
+func updateErrorStatePresent(report *LintReport, result ResultStruct) {
```
Comment (zakird) [inline lints/base.go]: not entirely sure why this is a function but fine

File: lints/base.go (hunk 1 above)
Comment (kumarde) [inline lints/base.go]: _sigh_, Yeah, I agree. The issue I was running into is that the results are abstracted away from the `ZLintResult` struct. I think what I can do is have the `LintReport` keep track of whether or not it has an error or warning or wahtever, and then bubble that up to a field in `ZLintResult`

Comment (zakird) [inline lints/base.go]: Huh?

File: lints/base.go (hunk 2 above)
Comment (zakird) [inline lints/base.go]: I'm not sure why this is here either. Why isn't all of this code in the protobuf thing that you generate given that's the location we need it?

File: lints/base.go (hunk 1 above)
Comment (kumarde) [inline lints/base.go]: I'll pass them up to the `ZLintResult`

File: lints/base.go (hunk 2 above)
Comment (kumarde) [inline lints/base.go]: To fill out the protobuf, we need to have this data available. I'd rather not iterate through every lint just to determine that there's an error or warning present, it's better if ZLint itself keeps track of that stuff and just exposes a bool to the caller

**Comment by kumarde** [comment]
@zakird mind taking a look at this whenever you get the chance? This needs to land before it gets updated in cert-daemon.

File: lints/base.go [hunk 3]
```diff
@@ -15,9 +15,13 @@ type lintReportUpdater func(*LintReport, ResultStruct)
 const ZLintVersion = 1
 
 type ZLintResult struct {
-	ZLintVersion int64
-	Timestamp    int64
-	ZLints       *LintReport
+	ZLintVersion   int64
+	Timestamp      int64
+	ZLints         *LintReport
+	NoticePresent  bool
```
Comment (zakird) [inline lints/base.go]: please use consistent pluralization.

File: lints/base.go [hunk 4]
```diff
@@ -207,6 +211,11 @@ type LintReport struct {
 	ERsaNoPublicKey                                      ResultStruct `json:"e_rsa_no_public_key,omitempty"`
 	ESubCertCertificatePoliciesMissing                   ResultStruct `json:"e_sub_cert_certificate_policies_missing,omitempty"`
 	ESubCertKeyUsageCrlSignBitSet                        ResultStruct `json:"e_sub_cert_key_usage_crl_sign_bit_set,omitempty"`
+	infosPresent                                         bool         `json:"infos_present,omitempty"`
```
Comment (zakird) [inline lints/base.go]: Why are there five vars here, but everywhere else in this PR, there are only four?

Comment (kumarde) [inline lints/base.go]: whoops.

File: lints/base.go (hunk 3 above)
Comment (kumarde) [inline lints/base.go]: done

File: lints/base.go [hunk 5]
```diff
@@ -207,6 +211,10 @@ type LintReport struct {
 	ERsaNoPublicKey                                      ResultStruct `json:"e_rsa_no_public_key,omitempty"`
 	ESubCertCertificatePoliciesMissing                   ResultStruct `json:"e_sub_cert_certificate_policies_missing,omitempty"`
 	ESubCertKeyUsageCrlSignBitSet                        ResultStruct `json:"e_sub_cert_key_usage_crl_sign_bit_set,omitempty"`
+	noticesPresent                                       bool         `json:"notices_present,omitempty"`
```
Comment (zakird) [inline lints/base.go]: This is not the right location for these to be output in the JSON.

Comment (kumarde) [inline lints/base.go]: fixed

**Pull request merged**